packages/host/app/lib/matrix-handlers/timeline.ts

```diff
@@ -9,11 +9,26 @@ import {
 
 import { eventDebounceMs } from '../matrix-utils';
 
-import { type Context, type Event, addRoomEvent } from './index';
+import {
+  type Context,
+  type Event,
+  addRoomEvent,
+  updateRoomEvent,
+} from './index';
 
 export function onTimeline(context: Context) {
   return (e: MatrixEvent) => {
-    context.timelineQueue.push(e);
+    context.timelineQueue.push({ event: e });
+    debouncedTimelineDrain(context);
+  };
+}
+
+export function onUpdateEventStatus(context: Context) {
+  return (e: MatrixEvent, _room: Room, oldEventId?: string) => {
+    if (typeof oldEventId !== 'string' || !e.status) {
```

Comment:
Why are you checking for the type of the `oldEventId`? Isn't the event id always a string?

Reply:
I am curious about this as well--what other type are we expecting here? if this is a fancy check for `undefined` or `null` then let's use:

```suggestion
    if (oldEvent == null || !e.status) {
```